avionics/common/src/sensors/battery.cpp

```diff
@@ -39,16 +40,21 @@ Battery::Battery(Pin batterySensorPin, float *const data)
 #endif
     m_divider = static_cast<float>(R2) / (R1 + R2);
     m_batterySensorPin = batterySensorPin;
-    status = SensorStatus::NOMINAL;
     // sets selected pin so it can be read from correctly
     Hal::pinMode(m_batterySensorPin, Hal::PinMode::INPUT);
+
+    if (getBatteryVoltage() < MINIMUM_BATTERY_VOLTAGE) {
+        status = SensorStatus::FAILURE;
+    } else {
+        status = SensorStatus::NOMINAL;
+    }
```

Comment:
Wondering what the practical difference is between this and the check in getBatteryVoltage() is? Do you think its good to keep both seperate? Or only do it in getBatteryVoltage() or put it in a separate function that both call?

Reply:
My intention was to only check `MINIMUM_BATTERY_VOLTAGE` and cause a critical failure when gathering the initial status but not inflight since I thought the focus of this task to prevent launching at all if the voltage was too low (my logic may be flawed here). 

Thinking about this again, there is a simple way to cause a noncritical failure inflight if the voltage is below `LOW_BATTERY_VOLTAGE` as well. 

Either way, I feel like the code would be clearer if these checks are separated because they check different conditions at different times, but we can discuss further.